central/discoveredclusters/datastore/datastore.go

```diff
@@ -0,0 +1,39 @@
+package datastore
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stackrox/rox/central/discoveredclusters/datastore/internal/search"
+	"github.com/stackrox/rox/central/discoveredclusters/datastore/internal/store"
+	pgStore "github.com/stackrox/rox/central/discoveredclusters/datastore/internal/store/postgres"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/discoveredclusters"
+	"github.com/stackrox/rox/pkg/postgres"
+)
+
+// DataStore is the entry point for modifying discovered cluster data.
+//
+//go:generate mockgen-wrapper
+type DataStore interface {
+	CountDiscoveredClusters(ctx context.Context, query *v1.Query) (int, error)
+	GetDiscoveredCluster(ctx context.Context, id string) (*storage.DiscoveredCluster, error)
+	ListDiscoveredClusters(ctx context.Context, query *v1.Query) ([]*storage.DiscoveredCluster, error)
+	UpsertDiscoveredClusters(ctx context.Context, discoveredClusters []*discoveredclusters.DiscoveredCluster) error
```

Comment:
nit: make this 
```suggestion
	UpsertDiscoveredClusters(ctx context.Context, discoveredClusters ...*discoveredclusters.DiscoveredCluster) error
```

so you can easily upsert single clusters as well without having to initialize an array.